educe/learning/edu_input_format.py

```diff
@@ -28,13 +31,16 @@ def _dump_edu_input_file(docs, f):
             edu_txt = edu.text().replace('\n', ' ')
             # subgroup: sentence identifier, backoff on EDU id
             sent_idx = edu2sent[i]
-            subgroup = ('{}_sent{}'.format(grouping, sent_idx)
-                        if sent_idx is not None
-                        else edu_gid)
+            if sent_idx is None:
+                subgroup = edu_gid
+            elif isinstance(sent_idx, six.string_types):
```

Comment:
What do you mean by 'natural string ID'?
Could you store turns in a list, then have edu2sent map from indices in the list of EDUs to indices in the list of turns?
(NB: edu2sent could be renamed edu2subgroup if we really wanted a more neutral terminology, but maybe overgeneralization is coming back here)